File: ypsopump/src/main/java/info/nightscout/androidaps/plugins/pump/ypsopump/defs/YpsoPumpCommandType.java
package info.nightscout.androidaps.plugins.pump.ypsopump.defs;

import androidx.annotation.StringRes;

import info.nightscout.androidaps.plugins.pump.ypsopump.R;

public enum YpsoPumpCommandType {

    SetBolus(R.string.ypsopump_cmd_desc_set_bolus),
    GetFirmwareVersion(R.string.ypsopump_cmd_desc_get_firmware),
    SetTemporaryBasalRate(R.string.ypsopump_cmd_desc_set_tbr),
    GetTemporaryBasalRate(R.string.ypsopump_cmd_desc_get_tbr),
    GetSettings(R.string.ypsopump_cmd_desc_get_settings),
    GetBasalProfile(R.string.ypsopump_cmd_desc_get_basal_profile),
    SetBasalProfile(R.string.ypsopump_cmd_desc_set_basal_profile)
    ;



    private Integer resourceId;

    YpsoPumpCommandType(@StringRes Integer resourceId) {
        this.resourceId = resourceId;
    }

    public Integer getResourceId() {
        return resourceId;
    }

}

File: ypsopump/src/main/java/info/nightscout/androidaps/plugins/pump/ypsopump/defs/YpsoPumpCommandType.kt
package info.nightscout.androidaps.plugins.pump.ypsopump.defs

import androidx.annotation.StringRes
import info.nightscout.androidaps.plugins.pump.ypsopump.R

enum class YpsoPumpCommandType(@param:StringRes val resourceId: Int) {
    SetBolus(R.string.ypsopump_cmd_desc_set_bolus),
    GetFirmwareVersion(R.string.ypsopump_cmd_desc_get_firmware),
    SetTemporaryBasal(R.string.ypsopump_cmd_desc_set_tbr),
    GetTemporaryBasal(R.string.ypsopump_cmd_desc_get_tbr),
    CancelTemporaryBasal(R.string.ypsopump_cmd_desc_cancel_tbr),
    GetSettings(R.string.ypsopump_cmd_desc_get_settings),
    GetBasalProfile(R.string.ypsopump_cmd_desc_get_basal_profile),
    SetBasalProfile(R.string.ypsopump_cmd_desc_set_basal_profile);

}